compiler/dyno/lib/resolution/resolution-queries.cpp

```diff
@@ -2002,6 +2072,57 @@ resolveFnCallForTypeCtor(Context* context,
   return mostSpecific;
 }
 
+using CandidatesVec = std::vector<const TypedFnSignature*>;
+
+static bool
+considerCompilerGeneratedCandidates(Context* context,
+                                    const CallInfo& ci,
+                                    const Scope* inScope,
+                                    const PoiScope* inPoiScope,
+                                    PoiInfo& poiInfo,
+                                    CandidatesVec& candidates) {
+
+  // only consider compiler-generated methods, for now
+  if (!ci.isMethod()) return false;
+
+  // fetch the receiver type info
+  assert(ci.numActuals() >= 1);
+  auto& receiver = ci.actuals(0);
+  auto receiverType = receiver.type().type();
+
+  // if not compiler-generated, then nothing to do
+  if (!needCompilerGeneratedMethod(context, receiverType, ci.name()))
+    return false;
+
+  // get the compiler-generated function, may be generic
+  auto tfs = getCompilerGeneratedMethod(context, receiverType, ci.name());
+  assert(tfs);
+
+  // check if the inital signature matches
+  auto faMap = FormalActualMap(tfs->untyped(), ci);
+  if (!isInitialTypedSignatureApplicable(context, tfs, faMap, ci)) {
+    return false;
+  }
+
+  // OK, already concrete, store and return
+  if (!tfs->needsInstantiation()) {
+    candidates.push_back(tfs);
+    return true;
+  }
+
+  // need to instantiate before storing
+  auto poi = pointOfInstantiationScope(context, inScope, inPoiScope);
+  auto instantiated = doIsCandidateApplicableInstantiating(context,
+                                                           tfs,
+                                                           ci,
+                                                           poi);
+
+  // TODO: also add POI info?
```

Comment:
I think you mean, TODO: also adjust poiInfo.

Seems to me you could make something based on this existing code:

```
  if (anyInstantiated) {
    instantiationPoiScope =
      pointOfInstantiationScope(context, inScope, inPoiScope);
    poiInfo.setPoiScope(instantiationPoiScope);
  }

  for (const TypedFnSignature* candidate : mostSpecific) {
    if (candidate != nullptr) {
      if (candidate->untyped()->idIsFunction()) {
        accumulatePoisUsedByResolvingBody(context, candidate,
                                          instantiationPoiScope, poiInfo);
      }
    }
  }
```